programs/aes/crypt_and_hash.c

```diff
@@ -172,7 +176,7 @@ int main( int argc, char *argv[] )
         mbedtls_fprintf( stderr, "Cipher '%s' not found\n", argv[4] );
         goto exit;
     }
-    if( ( ret = mbedtls_cipher_setup( &cipher_ctx, cipher_info) ) != 0 )
+    if( ( retval = mbedtls_cipher_setup( &cipher_ctx, cipher_info) ) != 0 )
```

Comment:
You're making an assignment to retval here, yet the variable is never referenced again.

I suggest rather than just remove the variable, split the assignment to `retval` and the test of the return value across two lines (easier to read), then output the value of `retval` if it fails.